src/analysis.h

```diff
@@ -736,58 +736,94 @@ namespace Faunus {
-                            idxfluc.push_back( ndx0+i );
-                            qavg.push_back( charge.at(i).avg() );
-                            qstdev.push_back( charge.at(i).stdev() );
-                            // we look for the id that was sampled most often
-                            auto id_max = std::max_element( std::begin(idcnt.at(i)), std::end(idcnt.at(i)), 
-                                [] (const std::pair<int, int>& p1, const std::pair<int, int>& p2) {
-                                    return p1.second < p2.second;
-                                }
-                            );
-                            mainid.push_back(atoms.at( id_max->first ).name);
+                        qavg.push_back( charge.at(i).avg() );
+                        qstdev.push_back( charge.at(i).stdev() );
+                        // we look for the id that was sampled most often
+                        auto id_max = std::max_element( std::begin(idcnt.at(i)), std::end(idcnt.at(i)), 
+                            [] (const std::pair<int, int>& p1, const std::pair<int, int>& p2) {
+                                return p1.second < p2.second;
                             }
+                        );
+                        mainname.push_back(atoms.at( id_max->first ).name);
+                    }
+                    if (verbose) {
+                        j[molname] = { {"main species", mainname},
+                            {"average q", qavg }, {"stdev of q", qstdev } };
+                    } else { 
+                        j["molecule"] = molname; 
                     }
-                    j = { {"indexes", idxfluc}, {"main species", mainid},
-                        {"average q", qavg }, {"stdev of q", qstdev } };
                 }
 
                 public:
                 ChargeFluctuations(const json &j, const Tspace &spc ) : spc(spc) {
                     from_json(j);
                     name = "chargefluctuations";
-                    if (j.count("indexes")==1) {
-                        indexes = j.value("indexes", decltype(indexes)());
-                        ndx0 = indexes.at(0);
-                        ndx1 = indexes.at(1);
-                    } else {
-                        ndx0 = 0 ;
-                        ndx1 = spc.p.size();
+                    verbose = j.value("verbose", false);
+                    molname = j.at("molecule").get<decltype(molname)>(); // molecule name
+                    auto it = findName(Faunus::molecules<Tpvec>, molname);
+                    if (it == Faunus::molecules<Tpvec>.end())
+                        throw std::runtime_error("unknown species '" + molname + "'");
+                    molid = it->id(); // molecule ID
+                    idcnt.resize( it->atoms.size() );
+                    charge.resize( it->atoms.size() );
+                }
+
+                virtual ~ChargeFluctuations() {
```

Comment:
cleaner to use the `FormatPQR` from io.h. I think it takes a vector of particles.